jujugui/static/gui/src/app/components/user-profile/user-profile.js

```diff
@@ -640,6 +652,66 @@ YUI.add('user-profile', function() {
         </div>);
     },
 
+    /**
+      Display the confirmation for destroying a model.
+
+      @method _displayConfirmation
+      @param {String} model The model to destroy.
+      @return {Object} The confirmation component.
+    */
+    _displayConfirmation: function(model) {
+      this.setState({destroyModel: model});
+    },
+
+    /**
+      Destroy a model.
+
+      @method _destroyModel
+      @param {String} uuid The model uuid to destroy.
+      @return {Object} The confirmation component.
+    */
+    _destroyModel: function(uuid) {
+      this._displayConfirmation(null);
+      this.props.destroyModel((error) => {
+        if (error) {
+          console.log(error);
```

Comment:
I've added a notification, not exactly sure how this will work with the model switching.